src/tls/connection.h

```diff
@@ -23,16 +23,29 @@
 
 #include "wsinstance.h"
 
+typedef enum { CLIENT, WS } DataSource;
+typedef enum { SUCCESS, PARTIAL, CLOSED, RETRY, FATAL } ConnectionResult;
+
+/* hold a block of read data that is being written */
+struct ConnectionBuffer {
+  int fd;
```

Comment:
I think it's confusing to put the fd into ConnectionBuffer struct, no?

Reply:
Maybe a little. The idea was to put the buffer and the fd into the same structure so that we can use epoll's user data pointer. However, we need a back reference to the parent Connection* object anyway, so if it makes things clearer I can move it back out. The reading part actually looks more elegant this way, but for the writing we need to take the respective other fd, which does look confusing.